operator/pkg/ciliumendpointslice/reconciler_test.go

```diff
@@ -258,3 +265,356 @@ func TestReconcileNoop(t *testing.T) {
 
 	hive.Stop(tlog, t.Context())
 }
+
+func TestReconcileCreate(t *testing.T) {
```

Comment:
It would be nice to convert the vast majority of higher level tests into script tests, which would be way easier to read, and allow for much better coverage testing the entire flow E2E. It can be a follow-up item though.